### PR TITLE
feat(e2e): added test for inputting ratings from progress/competency dashboard [SCHOOL-120]

### PR DESCRIPTION
**Pull request by Camus1859**

### CODE REVIEW

**Comment by themightychris** [comment]
@Camus1859 do you need any support with what you were testing here?

**Comment by Camus1859** [comment]
> @Camus1859 do you need any support with what you were testing here?

I didnt mean to add anyone to review it now. @nbey  and I are trying to schedule a time to walk through Sencha in detail and how to use it to grab the necessary elements @themightychris

**Comment by themightychris** [comment]
@Camus1859 please rename this PR

**Comment by holly-g** [comment]
@Camus1859 this should be ready to review, can you change the status from `draft`?

File: cypress/integration/comptency_dashboard_ratings.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                                        cy.contains('33%') // we input 1 rating for each skill (4/12 = 33%)
+                                        cy.contains('Y4') // Level 4 Progress
+                                        cy.contains('10') // Avg of 12
+                                    });
+
+
+                                // NOTE: the following assertions can fail if the fixture data changes
+                                // "12" rating on each skill in the first competency
+                                cy.get(`tr.cbl-grid-skills-row[data-competency="${firstCompetency.ID}"] [data-student="${studentId}"] li[title="12"]`)
+                                .should('have.length', firstCompetency.skillIds.length);
```
Comment (themightychris) [inline cypress/integration/comptency_dashboard_ratings.js]:
```suggestion
                                cy.get(`tr.cbl-grid-skills-row[data-competency="${firstCompetency.ID}"] [data-student="${studentId}"] li[title="12"]`)
                                    .should('have.length', firstCompetency.skillIds.length);
```

File: cypress/integration/comptency_dashboard_ratings.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+                        cy.get('#' + saveBtn.getId())
+                            .click({ force: true })
+
+                        cy.wait('@saveDemonstration')
+                            .should(({ xhr }) => {
+                                // confirm completion %, avg, and level
+                                const { data } = JSON.parse(xhr.response)
+                                const savedRecord = data[0];
+                                debugger;
```
Comment (themightychris) [inline cypress/integration/comptency_dashboard_ratings.js]:
```suggestion
```

File: cypress/integration/comptency_dashboard_ratings.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            cy.get('a[role=button] span.fa-plus').click()
+
+            // competency data loads when opening evidence submission form
+            cy.wait('@competencyData')
+                .should(({ xhr }) => {
+                    const { data: competencies, related: { ContentArea: contentAreas } } = JSON.parse(xhr.response);
+
+                    const firstCompetency = competencies[0];
+                    const secondCompetency = competencies[1];
+                    // const randomCompetency = competencies[competencies.length - Math.floor(Math.random() * (competencies.length - 2) + 2)];
```
Comment (themightychris) [inline cypress/integration/comptency_dashboard_ratings.js]: is this here for later? if so let's leave a note why, if not let's delete

File: cypress/integration/comptency_dashboard_ratings.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+            // get the 'Rubric' selector element
+            const rubricSelector = extQuerySelector('slate-cbl-contentareaselector')
+
+            // click the selector
+            cy.get('#' + rubricSelector.el.dom.id).click();
+
+            // verify and click first element of picker dropdown
+            cy.get('#' + rubricSelector.getPicker().id + ' .x-boundlist-item')
+            .contains('English Language Arts')
+            .click()
```
Comment (themightychris) [inline cypress/integration/comptency_dashboard_ratings.js]:
```suggestion
            cy.get('#' + rubricSelector.getPicker().id + ' .x-boundlist-item')
                .contains('English Language Arts')
                .click()
```

**Review by themightychris — CHANGES_REQUESTED** [review]
this test isn't passing right now, I shared some feedback in Slack: https://jarvus.slack.com/archives/C02A7LEFTFD/p1642448769001000

**Comment by themightychris** [comment]
This is actually looking really good after your recent work @nbey , looks to be failing within the new code:

Screen Shot 2022-02-22 at 2 49 13 PM

Also seems like there's an intermediate assertion we can add to be getting a better failure in this case

**Review by nbey — APPROVED** [review]

**Review by themightychris — APPROVED** [review]
@nbey @Camus1859 beautiful!

**Pull request merged**